src/detectChildProcessRule.ts

```diff
@@ -0,0 +1,128 @@
+import * as Lint from 'tslint';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import { AstUtils } from './utils/AstUtils';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+
+const FORBIDDEN_IMPORT_FAILURE_STRING: string = 'Found child_process import';
+const FOUND_EXEC_FAILURE_STRING: string = 'Found child_process.exec() with non-literal first argument';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'detect-child-process',
+        type: 'maintainability',
+        description: 'Tries to detect instances of child_process',
+        options: null, // tslint:disable-line:no-null-keyword
+        optionsDescription: '',
+        typescriptOnly: true,
+        issueClass: 'SDL',
+        issueType: 'Error',
+        severity: 'Important',
+        level: 'Opportunity for Excellence',
+        group: 'Security',
+        commonWeaknessEnumeration: '88'
+    };
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithFunction(sourceFile, walk);
+    }
+}
+
+function processImport(
+    ctx: Lint.WalkContext<void>,
+    node: ts.Node,
+    moduleAlias: string | undefined,
+    importedFunctionsAliases: string[],
+    importedModuleName: string,
+    childProcessModuleAliases: Set<string>,
+    childProcessFunctionAliases: Set<string>
+) {
+    if (importedModuleName === 'child_process') {
+        ctx.addFailureAt(node.getStart(), node.getWidth(), FORBIDDEN_IMPORT_FAILURE_STRING);
+        if (moduleAlias !== undefined) {
+            childProcessModuleAliases.add(moduleAlias);
+        }
+        importedFunctionsAliases.forEach(x => childProcessFunctionAliases.add(x));
+    }
+}
+
+function walk(ctx: Lint.WalkContext<void>) {
+    const childProcessModuleAliases = new Set<string>();
+    const childProcessFunctionAliases = new Set<string>();
+
+    function cb(node: ts.Node): void {
```

Comment:
This `cb` function is pretty large and getting a little hard to read. Could you please split the major parts of it into separate functions? They can be declared within the parent `walk` function if you'd like.

Example: `if (tsutils.isImportEqualsDeclaration(node)) {` declares variables within itself, so it's a good candidate for splitting out.

Reply:
Done